provisioning/Samples/service/BulkOperationSample/Program.cs

```diff
@@ -2,27 +2,34 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Reflection.Metadata.Ecma335;
+using BulkOperationSample;
+using CommandLine;
 
 namespace Microsoft.Azure.Devices.Provisioning.Service.Samples
 {
-    public class Program
+    internal class Program
     {
-        // The Provisioning Service connection string. This is available under the "Shared access policies" in the Azure portal.
-
-        // For this sample either:
-        // - pass this value as a command-prompt argument
-        // - set the PROVISIONING_CONNECTION_STRING environment variable 
-        // - create a launchSettings.json (see launchSettings.json.template) containing the variable
-        private static string s_connectionString = Environment.GetEnvironmentVariable("PROVISIONING_CONNECTION_STRING");
-
         public static int Main(string[] args)
         {
-            if (string.IsNullOrEmpty(s_connectionString) && args.Length > 0)
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+               .WithParsed(parsedParams =>
+               {
+                   parameters = parsedParams;
+               })
+               .WithNotParsed(errors =>
+               {
+                   Environment.Exit(1);
+               });
+

```

Comment:
Because the cs param is not "required" because it may be set via environment variable, you need to manually check that the value is not null or white space and use the AutoBuild thing that shows usage and exit the process.